File: Homeworks/26Jan/sequenceEquation.py
def permutationEquation(p: list):
    result = []
    for i in range(1, len(p) + 1):
        iIndex = p.index(i)
        result.append(p.index(iIndex + 1) + 1)
    
    return result

n = int(input())
p = [int(x) for x in input().split()]

print(permutationEquation(p))

File: Homeworks/26Jan/sequenceEquation.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> permutationEquation(vector<int> p) {
    vector<int> result;
    for (int i = 1; i <= static_cast<int>(p.size()); ++i) {
        int iIndex = distance(p.begin(), find(p.begin(), p.end(), i)) + 1;
        int y = distance(p.begin(), find(p.begin(), p.end(), iIndex)) + 1;
        result.push_back(y);
    }
    return result;
}


int main() {
    int n{}, v{};
    cin >> n;
    vector<int> p;
    for (int i = 0; i < n; ++i) {
        cin >> v;
        p.push_back(v);
    }
    for (auto value: permutationEquation(p)) {
        cout << value << " ";
    }
    cout << endl;
    return 0;
}